drivers/platform/chibios/chibios_uart.c

```diff
@@ -0,0 +1,191 @@
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include "no_os_error.h"
+#include "no_os_uart.h"
+#include "no_os_alloc.h"
+#include "chibios_uart.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Initialize the UART communication peripheral.
+ * @param desc  - The UART descriptor.
+ * @param param - The structure that contains the UART parameters.
+ * @return 0 in case of success, error code otherwise.
+ */
+static int32_t chibios_uart_init(struct no_os_uart_desc **desc,
+				 struct no_os_uart_init_param *param)
+{
+	struct chibios_uart_desc *chibios_uart;
+	struct chibios_uart_init_param *chibios_uart_ip;
+	struct no_os_uart_desc *descriptor;
+	int ret;
+	uint8_t data_bits;
+	uint8_t stop_bits;
+	uint8_t parity;
+
+	if (!desc || !param || !param->extra)
+		return -EINVAL;
+
+	descriptor = (struct no_os_uart_desc *) no_os_malloc(sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	chibios_uart = (struct chibios_uart_desc *) no_os_malloc(sizeof(*chibios_uart));
```

Comment:
is this variable passed to the no_os_uart_desc ? otherwise you need to free the memory at the end of the function regardless the result.